components/Common/ShellBox/index.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useCallback, useRef } from 'react';
+import { FormattedMessage } from 'react-intl';
+import styles from './index.module.scss';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+
+type Props = {
+  textToCopy?: string;
+};
+
+const ShellBox = ({ children, textToCopy }: React.PropsWithChildren<Props>) => {
+  const [copied, copyText] = useCopyToClipboard();
+
+  const shellBoxRef = useRef<HTMLElement>(null);
+
+  const handleCopyCode = useCallback(
+    async (event: React.MouseEvent<HTMLButtonElement>) => {
+      event.preventDefault();
+      const text = textToCopy || shellBoxRef.current?.innerHTML;
```

Comment:
Maybe we don't need a constant here, just `await copyText(textToCopy || shellBoxRef.current?.innerHTML);`

This also doesn't need to be an `useCallback` anymore as it doesn't do any complex computing anymore.